test/e2e/specs/onboarding/signup__with-theme-premium.ts

```diff
@@ -0,0 +1,218 @@
+/**
+ * @group calypso-release
+ */
+
+import {
+	DataHelper,
+	BrowserManager,
+	DomainSearchComponent,
+	UserSignupPage,
+	SignupPickPlanPage,
+	CartCheckoutPage,
+	StartSiteFlow,
+	SecretsManager,
+	NewSiteResponse,
+	RestAPIClient,
+	NewUserResponse,
+	MyProfilePage,
+	MeSidebarComponent,
+	cancelPurchaseFlow,
+	NoticeComponent,
+	PurchasesPage,
+	ThemesDetailPage,
+} from '@automattic/calypso-e2e';
+import { Page, Browser } from 'playwright';
+import { apiCloseAccount } from '../shared';
+
+declare const browser: Browser;
+
+/**
+ * Checks the entire with theme user lifecycle, from signup, onboarding, launch and plan cancellation.
+ *
+ * Keywords: Onboarding, Store Checkout, Coupon, Signup, Plan, Subscription, Cancel
+ */
+describe( 'With theme - Premium theme: Signup, onboard, launch and cancel subscription', function () {
+	const planName = 'Premium';
+	const themeSlug = 'outland';
+
+	const testUser = DataHelper.getNewTestUser( {
+		usernamePrefix: 'ftmepersonal',
+	} );
+
+	let page: Page;
+	let newUserDetails: NewUserResponse;
+	let newSiteDetails: NewSiteResponse;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+	} );
+
+	describe( 'Signup', function () {
+		let cartCheckoutPage: CartCheckoutPage;
+		let signupPickPlanPage: SignupPickPlanPage;
+		let originalAmount: number;
+
+		beforeAll( async function () {
+			await BrowserManager.setStoreCookie( page, { currency: 'GBP' } );
+		} );
+
+		it( 'Navigate to Signup page', async function () {
+			const themeDetailsPage = new ThemesDetailPage( page );
+			await themeDetailsPage.visitTheme( themeSlug );
+
+			await themeDetailsPage.pickThisDesign();
+		} );
+
+		it( 'Sign up as new user', async function () {
+			const userSignupPage = new UserSignupPage( page );
+			newUserDetails = await userSignupPage.signupSocialFirstWithEmail( testUser.email );
+		} );
+
+		it( 'Skip domain selection', async function () {
+			const domainSearch = new DomainSearchComponent( page );
+
+			await domainSearch.search( testUser.siteName );
+			await domainSearch.selectDomain( `${ testUser.siteName }.wordpress.com` );
```

Comment:
**suggestion:non-blocking**- I'm going to actually suggest that, to make this spec more useful, we can cover the scenario of explicitly selecting a Free WordPress.com domain. What do you think?

Reply:
I avoided selecting a domain name because I didn't want to risk actually registering an actual domain name.